src/Text/Pandoc/App/Opt.hs

```diff
@@ -475,16 +478,14 @@ defaultOpts = Opt
     , optStripComments         = False
     }
 
-contextToMeta :: Context Text -> Meta
-contextToMeta (Context m) =
-  Meta . M.map valToMetaVal $ m
-
-valToMetaVal :: Val Text -> MetaValue
-valToMetaVal (MapVal (Context m)) =
-  MetaMap . M.map valToMetaVal $ m
-valToMetaVal (ListVal xs) = MetaList $ map valToMetaVal xs
-valToMetaVal (SimpleVal d) = MetaString $ render Nothing d
-valToMetaVal NullVal = MetaString ""
+yamlToMeta :: Node Pos -> Meta
+yamlToMeta (Mapping _ _ m) = runEverything (yamlMap pMetaString m)
+  where
+    pMetaString = pure . MetaString <$> P.manyChar P.anyChar
+    runEverything :: P.ParserT Text P.ParserState PandocPure (P.F (M.Map Text MetaValue)) -> Meta
+    runEverything p =
+      either (const mempty) (Meta . flip P.runF def) . join . runPure $ P.readWithM p def ""
+yamlToMeta _ = mempty
```

Comment:
I'm confused about this code. What's the point of `runEverything` here if we're just passing in `pMetaString`?

Reply:
Basically: to make the type checker happy. `yamlMap` is several monad stacks deep before we get to the actualy `MetaMap`, `runEverything` has to unwrap those.

FWIW, some of those layers need to be joined with the `Parser` in `doOpt` for the purposes of error handling, but I was being a little lazy with the proof-of-concept.